src/main/java/core/basesyntax/RightTriangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class RightTriangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final ColorSupplier color;
```

Comment:
The `ColorSupplier` should not be a field in the `RightTriangle` class. Instead, the color should be a property of the `Figure` class, and it should be of type `Color`, not `ColorSupplier`.